homestar-runtime/src/workflow/info.rs

```diff
@@ -348,26 +359,17 @@ impl Info {
                     "workflow information not available in the database"
                 );
 
-                Self::retrieve_from_query(
-                    workflow_cid,
-                    p2p_timeout,
-                    event_sender,
-                    conn,
-                    handle_timeout_fn,
-                )
-                .await
+                Self::retrieve_from_provider(workflow_cid, event_sender, p2p_provider_timeout).await
```

Comment:
@bgins let's comment this out for now and put a TODO to return to with for the recursive case. We should also re-explore how we do call gather in `swarm_event` and probably link a TODO on there too.

Reply:
I'll update this in a follow-up.